File: src/main/java/com/rabit/api/dto/response/CompanyUserResponse.java
package com.rabit.api.dto.response;

import com.rabit.api.model.enums.CompanyRole;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.Instant;
import java.util.UUID;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class CompanyUserResponse {

    private UUID userId;

    private String email;

    private String firstName;

    private String lastName;

    private CompanyRole role;

    private boolean active;

    private Instant createdAt;

    private String message;
}

File: dtos/responses/CompanyUserResponse.js
const { toRoleSummary } = require('../helpers/RoleSummary');

class CompanyUserResponse {
  constructor(companyUser) {
    this.id = companyUser.id;
    this.userId = companyUser.userId;
    this.companyId = companyUser.companyId;
    this.active = companyUser.active;
    this.createdAt = companyUser.createdAt;
    this.user = companyUser.user ? {
      userId: companyUser.user.userId,
      username: companyUser.user.username,
      email: companyUser.user.email,
      firstName: companyUser.user.firstName || null,
      lastName: companyUser.user.lastName || null,
      status: companyUser.user.status,
    } : null;
    this.role = toRoleSummary(companyUser.role);
  }
}
module.exports = CompanyUserResponse;
